app/controllers/white_cards.rb

```diff
@@ -0,0 +1,40 @@
+
+get '/white_cards' do
+  if !logged_in?
+    erb :'/please_sign_in'
```

Comment:
You should redirect to your login page rather than rendering this at this url. (same for all the other places you're doing this)

Also you can make a helper called `ensure_logged_in!` that uses Sinatra's `halt` feature to redirect for any route that required you be logged in.